azurerm/internal/services/netapp/resource_arm_netapp_volume.go

```diff
@@ -89,6 +90,19 @@ func resourceArmNetAppVolume() *schema.Resource {
 				ValidateFunc: azure.ValidateResourceID,
 			},
 
+			"protocols": {
+				Type:     schema.TypeList,
```

Comment:
This should be a type set

Reply:
Fixed.

Reply:
Hi @katbyte , any idea why when I changed this to schema.TypeSet I started getting validation error on my test?

--- FAIL: TestAccAzureRMNetAppVolume_nfsv41 (425.90s)
    testing.go:569: Step 0 error: Check failed: Check 2/2 error: azurerm_netapp_volume.test: Attribute 'protocols.0' not found
FAIL

Here is snippet of the test:

```go
func TestAccAzureRMNetAppVolume_nfsv41(t *testing.T) {
	data := acceptance.BuildTestData(t, "azurerm_netapp_volume", "test")

	resource.ParallelTest(t, resource.TestCase{
		PreCheck:     func() { acceptance.PreCheck(t) },
		Providers:    acceptance.SupportedProviders,
		CheckDestroy: testCheckAzureRMNetAppVolumeDestroy,
		Steps: []resource.TestStep{
			{
				Config: testAccAzureRMNetAppVolume_nfsv41(data),
				Check: resource.ComposeTestCheckFunc(
					testCheckAzureRMNetAppVolumeExists(data.ResourceName),
					resource.TestCheckResourceAttr(data.ResourceName, "protocols.0", "NFSv4.1"),
				),
			},
			data.ImportStep(),
		},
	})
}
```

Reply:
I think I solved it, I used the hash that the Set generates and I don't get the error as follows:

```go
resource.TestCheckResourceAttr(data.ResourceName, "protocols.3098200649", "NFSv4.1"),
```